src/Microsoft.VisualStudio.ProjectSystem.Managed/Tree/Dependencies/Subscriptions/DependencySharedProjectsSubscriber.cs

```diff
@@ -56,12 +58,10 @@ public void AddSubscriptions(AggregateCrossTargetProjectContext projectContext)
 
         public void ReleaseSubscriptions()
         {
-            foreach (IDisposable link in _subscriptionLinks)
-            {
-                link.Dispose();
-            }
+            // Prevent double-dispose
+            DisposableBag? value = Interlocked.Exchange(ref _subscriptions, null);
```

Comment:
That's exactly what DisposableBag does under the covers to prevent double-disposing.

Reply:
The comment is perhaps incomplete. DisposableBag is a one-time thing. We need to recreate the instance for future subscriptions, done in another interlocked operation elsewhere in this PR.

Reply:
I removed the interlocked stuff here as it was incomplete and the caller is currently synchronising calls anyway.

This will be revisited as I work towards consolidating `DependencyRulesSubscriber` and `DependencySharedProjectsSubscriber` which both contain similar code.